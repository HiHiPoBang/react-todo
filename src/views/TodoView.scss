.TodoView{
  width: 100vw;
  max-width: 100vw;
  min-height: 100vh;
  overflow-x: hidden;
  &__main {
    display: flex;
    align-items: flex-start;
    justify-content: center;
    padding: 20px 0px;
  }
  &__column {
    display: grid;
    grid-template-columns: 1fr;
    align-items: flex-start;
    justify-content: flex-start;
    grid-row-gap: 1rem;
    padding: 0.5rem
  }
  &__menu {
    position: fixed;
    bottom: 10px;
    right: 10px;
    margin: 10px;
    padding: 0;
    list-style-type: none;
  }
}