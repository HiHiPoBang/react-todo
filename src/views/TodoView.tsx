import React, { useState } from 'react';
import * as R from 'ramda';
import { nanoid } from 'nanoid';
import './TodoView.scss';
import { TodoObjInterface } from '../components/TodoList/todoUtils';
import TodoList from '../components/TodoList/Todo';
import { StyledButton } from '../styled-components';

const listData = [
  {
    id: nanoid(), label: 'hello', type: 'danger',
  },
  {
    id: nanoid(), label: 'hello', type: 'warning',
  },
];
const todosData = [{ todoId: nanoid(), title: 'Untitled Todo', todo: [...listData] }];
interface todoItemParam {
  label: string,
  type: string,
}
interface deleteTodoParam {
  todoId: string,
  id: string,
}
const TodoView = function () {
  const [todos, setTodos] = useState(todosData);
  const createNewTodoItem = (val: todoItemParam) => R.merge({ id: nanoid() }, val);
  const createNewTodo = (val: TodoObjInterface) => ({
    todoId: nanoid(),
    title: 'Untitled todo',
    todo: [val],
  });
  const deleteTodoItem = (param: deleteTodoParam) => {
    const { todoId, id } = param;
  };
  const addTodoItem = (param: todoItemParam, todoIdx: number) => {
    const todoObj = todos[todoIdx];

    R.pipe(
      (val) => createNewTodoItem(val),
      (newItem) => ({ ...todoObj, todo: [...todoObj.todo, newItem] }),
      (newList) => R.update(todoIdx, newList, todos),
      (newTodos) => setTodos(newTodos),
    )(param);
  };
  const addNewTodo = () => R.pipe(
    (param) => createNewTodoItem(param),
    (newItem) => createNewTodo(newItem),
    (newTodo) => setTodos([...todos, newTodo]),
  )({ label: 'untitled item', type: 'default' });
  return (
    <div className="TodoView">
      {
        todos.map((todoObj, index) => (
          <TodoList
            key={todoObj.todoId}
            list={todoObj.todo}
            addTodoItem={(val: todoItemParam) => addTodoItem(val, index)}
          />
        ))
      }
      <div>
        <StyledButton onClick={addNewTodo}>Add new todo</StyledButton>
      </div>
    </div>
  );
};

export default TodoView;
