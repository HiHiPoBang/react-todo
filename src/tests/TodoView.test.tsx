import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import TodoView from '../views/TodoView';

describe('Rendered TodoView', () => {
  test('ShallowMount TodoView ', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<TodoView />);
    const result = renderer.getRenderOutput();
    expect(result.type).toBe('div');
    expect(result.props.className).toEqual('TodoView');
  });
  test('Create snapshot', () => {
    const snapshot = create(<TodoView />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
