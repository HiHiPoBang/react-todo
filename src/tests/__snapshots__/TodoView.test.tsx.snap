// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered TodoView Create snapshot 1`] = `
.c2 {
  font-size: 0.875rem;
}

.c0 {
  font-size: 0.875rem;
}

.c1 {
  margin: 15px 0px;
  font-size: 2rem;
  color: #333333;
}

<div
  className="TodoView"
>
  <div
    className="TodoList"
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      X
    </button>
    <h2
      className="c1"
    >
      My Todo
    </h2>
    <ul
      className="TodoList__list"
    >
      <li
        className="danger"
      >
        <div>
          <span>
            hello
          </span>
          <button
            className="c0"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
      </li>
      <li
        className="warning"
      >
        <div>
          <span>
            hello
          </span>
          <button
            className="c0"
            onClick={[Function]}
          >
            delete
          </button>
        </div>
      </li>
    </ul>
    <form
      onSubmit={[Function]}
    >
      <input
        className="c2"
        onChange={[Function]}
        placeholder="Unlabeld"
        value=""
      />
      <button
        className="c0"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <div>
    <button
      className="c0"
      onClick={[Function]}
    >
      Add new todo
    </button>
  </div>
</div>
`;
